tests/test_magnetic_fields.py

```diff
@@ -1201,25 +1201,35 @@ def test_field_line_integrate(self):
         np.testing.assert_allclose(z[-1], 0.001, rtol=1e-6, atol=1e-6)
 
     @pytest.mark.unit
-    def test_field_line_integrate_jax_transforms(self):
+    def test_field_line_integrate_jax_transforms(self, capsys):
         """Test field line integration JAX transformable."""
         field = ToroidalMagneticField(2, 10) + PoloidalMagneticField(2, 10, 0.25)
         r0 = [10.001]
         z0 = [0.0]
         phis = [0, 2 * np.pi]
+
+        def fun0(r0, z0, field):
+            return field_line_integrate(r0, z0, phis, field, max_steps=1000)
+
         # check if it is jittable
-        r, z = jit(field_line_integrate)(r0, z0, phis, field)
+        r, z = jit(fun0)(r0, z0, field)
```

Comment:
i think you don't want to jit fun0 here if you're trying to check recompilation. This should pass on master I think? since its only caching based on r0, z0 and field.